app/services/school_data_importer.rb

```diff
@@ -0,0 +1,70 @@
+require "open-uri"
+require "csv"
+
+class SchoolDataImporter
+  def run
+    download_and_save_csv
+
+    CSV.foreach(csv_file_name, headers: true, encoding: "windows-1251:utf-8") do |row|
+      local_authority = LocalAuthority.find_or_create_by(
+        code: row["LA (code)"],
+        name: row["LA (name)"]
+      )
+
+      school = row_to_school(row)
+
+      school.local_authority = local_authority
```

Comment:
how about moving the finding/creating of the school and this assignment into the `row_to_school` method to make things more encapsulated.